tests/unit/nupic/research/spatial_pooler_boost_test.py

```diff
@@ -339,13 +359,49 @@ def boostTestLoop(self, imp):
     self.boostTestPhase3()
     self.boostTestPhase4()
 
+
   def testBoostingPY(self):
-    self.boostTestLoop("py")
+    self.setUp(imp="py")
+    self.boostTestLoop()
 
 
   def testBoostingCPP(self):
-    self.boostTestLoop("cpp")
-
+    self.setUp(imp="cpp")
+    self.boostTestLoop()
+
+
+  def testBoostingDefaultSPParams(self):
+    """Boosting test with SP's default params."""
+    params = {}
+    params['seed'] = SEED
+    # py
+    self.setUp(params, "py") # reinitialize with new params
+    self.boostTestLoop()
+    # cpp
+    self.setUp(params, "cpp")
+    self.boostTestLoop()
+
+
+  def testBoostingSDRPatterns(self):
+    """Boosting test with random patterns, instead of continuous data."""
```

Comment:
It's not clear what this is testing from this docstring. The patterns are not really random, since the first two are identical and the third one shares 50% of the bits. Can you elaborate here on exactly what this new test is testing?